vllm/model_executor/layers/fused_moe/deep_gemm_utils.py

```diff
@@ -0,0 +1,415 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+"""
+Taken from https://github.com/ModelTC/lightllm/blob/main/lightllm/common/fused_moe/deepep_scatter_gather.py 
```

Comment:
nit: better to permalink rather than pointing to main so we can track changes better if it drifts